openlibrary/macros/ShareModal.html

```diff
@@ -51,7 +51,9 @@
             </li>
 
             <li>
-              <a class="copy-url-btn" title="$_('Copy url to clipboard')" data-ol-link-track="Share|CopyURL">
+              $ full_url = 'https://' + ctx.site + changequery()
```

Comment:
**Reviewer:** `ctx.site` is `openlibrary.org` in my local development instance.  Will this be the same on `testing` and `staging`?  If so, how much does it matter for this particular case?

Reply:
[`Request`](https://github.com/internetarchive/openlibrary/blob/c1e1dfe3a3a1f5fc92755f7fd5ffee52df83cfe7/openlibrary/plugins/upstream/utils.py#L1507) object is accessible via the template.  Use `canonical_url` here.

Reply:
Above in the iframe we use `https:$request.domain$page_url` ; maybe that and adding `changequery()` to that? `ctx.site` seems to be `openlibrary.org` even on testing.